Framework/Parallel/src/StorageMode.cpp

```diff
@@ -25,5 +25,18 @@ std::string toString(const std::map<std::string, StorageMode> &map) {
   return ret;
 }
 
+/// Returns a StorageMode for a human-readable string representation.
+StorageMode fromString(const std::string &mode) {
+  if (mode == "Parallel::StorageMode::Cloned")
```

Comment:
Not sure about the `Parallel::` part, but `StorageMode::` is included to distinguish it from `ExecutionMode` (similar to having `StorageMode` and `ExecutionMode` as strongly typed enums).